prow/plugins/trigger/pull-request.go

```diff
@@ -106,6 +106,45 @@ func handlePR(c Client, trigger plugins.Trigger, pr github.PullRequestEvent) err
 				return buildAll(c, &pr.PullRequest, pr.GUID, trigger.ElideSkippedContexts)
 			}
 		}
+		if pr.Label.Name == labels.OkToTest {
+			labelAddedBy := pr.Sender.Name
+			// If label is added by the bot, then skip it.
+			// this could also be possible if someone goes the traditional way of adding /ok-to-test then /test-all comments
+			// to trigger the testing
+			botName, err := c.GitHubClient.BotName()
+			if err != nil {
+				return err
+			}
+			if labelAddedBy == botName {
+				c.Logger.Debug("Label added by the bot, skipping.")
+				return nil
+			}
+			trusted, err := TrustedUser(c.GitHubClient, trigger.OnlyOrgMembers, trigger.TrustedOrg, labelAddedBy, org, repo)
+			if err != nil {
+				return fmt.Errorf("could not validate user: %s", err)
+			}
+
+			if !trusted {
+				return fmt.Errorf("%s is an untrusted user", labelAddedBy)
+			}
+
+			var filters []pjutil.Filter
+			filters = append(filters, pjutil.TestAllFilter())
+
+			changes := config.NewGitHubDeferredChangedFilesProvider(c.GitHubClient, org, repo, pr.PullRequest.Number)
+
+			pullRequest, err := c.GitHubClient.GetPullRequest(org, repo, pr.Number)
```

Comment:
You do not need to fetch the PullRequest, its part of the PullRequestEvent this funtion gets called with.

Also, you can use `buildAll` here so you do not have to do all the Presubmit filtering by yourself